cmd/drone-server/server.go

```diff
@@ -675,6 +675,12 @@ func (a *authorizer) authorize(ctx context.Context) error {
 	return errors.New("missing agent token")
 }
 
+func redirect(w http.ResponseWriter, req *http.Request) {
+	req.URL.Scheme = "https"
+	req.URL.Host = req.Host
```

Comment:
Maybe it's better to set this with the parse result of the server-host flag.

Reply:
In general, I would prefer to be redirected to the host the request was made to (mostly for debugging, being able to go to `drone-server001.company.com` directly instead of `drone.company.com` is often useful). Given the oauth flow redirects back the the server-host anyways, I can see why redirecting here would be desired. Will update!

Reply:
But this potentially lead to invalid certificates, especially if let's encrypt is enabled.